internal/services/cdn/cdn_frontdoor_firewall_policy_resource.go

```diff
@@ -623,74 +627,75 @@ func resourceCdnFrontDoorFirewallPolicyUpdate(d *pluginsdk.ResourceData, meta in
 
 	if d.HasChange("tags") {
 		t := d.Get("tags").(map[string]interface{})
-		existing.Tags = expandFrontDoorTags(tags.Expand(t))
+		model.Tags = tags.Expand(t)
 	}
 
-	existing.WebApplicationFirewallPolicyProperties = &props
-	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.FrontDoorWebApplicationFirewallPolicyName, existing)
+	model.Properties = pointer.To(props)
+
+	err = client.PoliciesCreateOrUpdateThenPoll(ctx, pointer.From(id), pointer.From(model))
 	if err != nil {
 		return fmt.Errorf("updating %s: %+v", *id, err)
 	}
-	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		return fmt.Errorf("waiting for the update of %s: %+v", *id, err)
-	}
 
 	return resourceCdnFrontDoorFirewallPolicyRead(d, meta)
 }
 
 func resourceCdnFrontDoorFirewallPolicyRead(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Cdn.FrontDoorLegacyFirewallPoliciesClient
+	client := meta.(*clients.Client).Cdn.FrontDoorFirewallPoliciesClient
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.FrontDoorFirewallPolicyID(d.Id())
+	id, err := waf.ParseFrontDoorWebApplicationFirewallPolicyID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.FrontDoorWebApplicationFirewallPolicyName)
+	result, err := client.PoliciesGet(ctx, pointer.From(id))
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
-			log.Printf("[INFO] Cdn Frontdoor Firewall Policy %q does not exist - removing from state", d.Id())
-			d.SetId("")
-			return nil
-		}
 		return fmt.Errorf("retrieving %s: %+v", *id, err)
 	}
 
+	model := result.Model
+
+	if model == nil {
+		return fmt.Errorf("retrieving %s: 'model' was nil", *id)
+	}
+
+	if model.Properties == nil {
+		return fmt.Errorf("retrieving %s: 'model.Properties' was nil", *id)
+	}
```

Comment:
Think I missed this on the last pass? Can we use the `if model := result.Model; model != nil {}` pattern here for the read? The `d.Set()` items from the ID can be performed outside this, ofc.

Reply:
Fixed.